components/Author/Tabs/FeedCard.tsx

```diff
@@ -0,0 +1,558 @@
+import { emptyFncWithMsg, isNullOrUndefined } from "~/config/utils/nullchecks";
+import { formatDate } from "~/config/utils/dates";
+import { isDevEnv } from "~/config/utils/env";
+import { transformDate } from "~/redux/utils";
+
+const PaperPDFModal = dynamic(
+  () => import("~/components/Modals/PaperPDFModal")
+);
+
+export type FeedCardProps = {
+  boost_amount: number;
+  created_by: any;
+  created_date: any;
+  discussion_count: number;
+  first_figure: any;
+  first_preview: any;
+  formattedDocType: string | null;
+  hubs: any[];
+  id: number;
+  index: number;
+  onBadgeClick: any;
+  openPaperPDFModal: any;
+  paper: any;
+  postDownvote: any;
+  postUpvote: any;
+  preview_img: string;
+  renderableTextAsHtml: any;
+  renderable_text: string;
+  score: number;
+  singleCard: boolean;
+  slug: string;
+  style: StyleSheet;
+  styleVariation: string;
+  title: string;
+  titleAsHtml: any;
+  unified_document: any;
+  unified_document_id: number;
+  uploaded_by: any;
+  uploaded_date: any;
+  user: any;
+  user_vote: any;
+  voteCallback: any;
+};
+
+function FeedCard(props: FeedCardProps) {
+  const {
+    boost_amount: boostAmount,
+    created_by,
+    created_date,
+    discussion_count,
+    first_figure,
+    first_preview,
+    formattedDocType,
+    hubs,
+    id,
+    index,
+    onBadgeClick,
+    paper,
+    postDownvote,
+    postUpvote,
+    preview_img: previewImg,
+    renderableTextAsHtml,
+    renderable_text: renderableText,
+    score: initialScore,
+    singleCard,
+    slug,
+    style,
+    styleVariation,
+    title,
+    titleAsHtml, // In some contexts we want to wrap the title/renderable_text with html. e.g. rendering search highlights.
+    unified_document: unifiedDocument,
+    unified_document_id: unifiedDocumentId,
+    uploaded_by,
+    uploaded_date,
+    user: currentUser,
+    user_vote: userVote,
+    voteCallback,
+  } = props;
+
+  /**
+   * Whether or not THIS PaperPDFModal is open.
+   * There may be many PaperPDFModal components on the page, but
+   * modals.openPaperPDFModal is only a single boolean. So all cards
+   * must only render their PaperPDFModal component if requested */
+  const [isPreviewing, setIsPreviewing] = useState(false);
+  const [voteState, setVoteState] = useState<string | null>(
+    userVoteToConstant(userVote)
+  );
+  const [score, setScore] = useState<number>(initialScore + (boostAmount || 0));
+  const [isHubsOpen, setIsHubsOpen] = useState(false);
+  const [previews] = useState(
+    configurePreview([
+      first_preview && first_preview,
+      first_figure && first_figure,
+    ])
+  );
+
+  useEffect((): void => {
+    setVoteState(userVoteToConstant(userVote));
+  }, [userVote]);
+
+  function configurePreview(arr) {
+    return arr.filter((el) => {
+      return !isNullOrUndefined(el);
+    });
+  }
+
+  const createVoteHandler = (voteType) => {
+    const voteStrategies = {
+      [UPVOTE]: {
+        increment: 1,
+        getUrl:
```

Comment:
Maybe I'm missing something but this seem to only imply a voting strategy on posts and hypotheses but what about papers?

Reply:
Papers have a separate callback called `onPaperVote`, this was legacy logic from PaperEntryCard I wasn't confident in changing.